File: tracker/src/test/java/testhelpers/JUnitTree.java
package testhelpers;

import androidx.annotation.NonNull;
import android.util.Log;

import timber.log.Timber;


public class JUnitTree extends Timber.DebugTree {
    private final int minlogLevel;

    public JUnitTree() {
        minlogLevel = Log.VERBOSE;
    }

    private static String priorityToString(int priority) {
        switch (priority) {
            case Log.ERROR:
                return "E";
            case Log.WARN:
                return "W";
            case Log.INFO:
                return "I";
            case Log.DEBUG:
                return "D";
            case Log.VERBOSE:
                return "V";
            default:
                return String.valueOf(priority);
        }
    }

    @Override
    protected void log(int priority, String tag, @NonNull String message, Throwable t) {
        if (priority < minlogLevel) return;
        System.out.println(System.currentTimeMillis() + " " + priorityToString(priority) + "/" + tag + ": " + message);
    }
}

File: tracker/src/test/java/testhelpers/JUnitTree.kt
package testhelpers

import android.util.Log
import timber.log.Timber

class JUnitTree : Timber.DebugTree() {
    private val minLogLevel = Log.VERBOSE

    override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
        if (priority < minLogLevel) return
        println(System.currentTimeMillis().toString() + " " + priorityToString(priority) + "/" + tag + ": " + message)
    }

    companion object {
        private fun priorityToString(priority: Int): String {
            return when (priority) {
                Log.ERROR -> "E"
                Log.WARN -> "W"
                Log.INFO -> "I"
                Log.DEBUG -> "D"
                Log.VERBOSE -> "V"
                else -> priority.toString()
            }
        }
    }
}
